File: atlas/examples/kotlin/app/src/main/kotlin/dev/atlas/example/app/RuntimeOutputWriter.kt
package dev.atlas.example.app

/**
 * Writes the application's designed user-facing output to the runtime stream.
 */
class RuntimeOutputWriter {
    /**
     * Writes one complete user-facing line.
     *
     * @param message Rendered content without a trailing newline.
     */
    fun writeLine(message: String) {
        println(message)
    }
}

File: atlas/examples/typescript/packages/app/src/RuntimeOutputWriter.ts
/**
 * Writes the application's designed user-facing output to the runtime stream.
 */
export class RuntimeOutputWriter {
  /**
   * Writes one complete user-facing line.
   *
   * @param message - Rendered content without a trailing newline.
   */
  public writeLine(message: string): void {
    stdout.write(`${message}\n`);
  }
}
import { stdout } from 'node:process';
